postgresql-11.5/src/fe_utils/printbteq.c

```diff
@@ -922,20 +957,50 @@ print_aligned_text_bteq(const printTableContentBteq *cont, FILE *fout, bool is_p
                 {
                     struct lineptr *this_line = col_lineptrs[i] + curr_nl_line;
                     unsigned int nbspace;
+                    if (cont->table_width) {
+                        int border_c = (i+1 != cont->ncolumns ? 1 : 0);
+                        if ((width_wrap[i]+2 + border_c) <= table_width) {
+                            table_width -= (width_wrap[i] + 2 + border_c);
+                            width_wrap_copy[i] = width_wrap[i] + 2;
+                        } else {
+                            table_width2 = table_width;
+                            int nbspace_r = 0;
+                            nbspace = width_wrap[i] - this_line->width + 2;
+                            width_wrap_copy[i] = table_width;
+                            col_count_copy = i + 1;
+                            if (nbspace / 2 >= table_width ) {
+                                nbspace = table_width * 2;
+                                this_line->ptr[0] = '\0';
+                                nbspace_r = 0;
+                            } else {
+                                table_width -= (nbspace / 2);
+                                if (this_line->width >= table_width) {
+                                    this_line->ptr[table_width] = '\0';
+                                    nbspace_r = 0;
+                                } else {
+                                    nbspace_r = (table_width - this_line->width) * 2;
+                                }
+                            }
+
+                            fprintf(fout, "%-*s%s%-*s",
+                                    (nbspace ) / 2, "", this_line->ptr,
+                                    (nbspace_r+1) / 2, "");
+
+                            more_col_wrapping = 0;
+                            goto loop_end;
+                        }
+                    }
 
                     if (opt_border != 0 ||
                         (!format->wrap_right_border && i > 0))
                         fputs(curr_nl_line ? format->header_nl_left : " ",
                               fout);
-
```

Comment:
do not need to remove the new lines.

Reply:
done as specified

Reply:
I still can see newlines removed. Actually - not done

Reply:
Maybe you didn't commit the changes?